Sources/Plasma/PubUtilLib/plGImage/plPNG.cpp

```diff
@@ -226,27 +229,59 @@ bool plPNG::IWrite(plMipmap* source, hsStream* outStream)
             row_ptrs[i] = (png_bytep)srcp + (i * stride);
         }
 
+        // Write image data
         png_write_image(png_ptr, row_ptrs);
+
+        // Prepare Textual Metadata
+        bool owned_map = false;
+        if (!textFields) {
+            auto textFields = new std::multimap<plString, plString>();
```

Comment:
Are you sure?  I made sure to clean it up at the end, and only if we created the inner one.  A bit messy, and the one part I wasn't happy about, but it all goes away now anyway.